tests/unit_tests/arelle/test_validatexbrlcalcs.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+import pytest
+from _decimal import Decimal, InvalidOperation
+
+from arelle.ValidateXbrlCalcs import insignificantDigits
+
+
+@pytest.mark.parametrize('value, precision, decimals, scale, result, error', [
+    # precision
+    (1234, -1, None, None, ('0', '1234'), None),
```

Comment:
That may be the case, but the current implementation of the function makes no special consideration for negative precision values, and even attempts to handle them as one might expect mathematically, so I'm of the mindset that we test for it. Especially since it's an established Arelle function with users not necessarily visible to us (core Arelle doesn't even make use of `precision` or `scale` arguments).

Now, if we want the expectation of non-negative precision values to be represented in the implementation of the function, that's another conversation to have (I could include that change in my next `insignificantDigits` PR), but we still benefit from having a unit test case that will catch the corresponding changes to the results.

Reply:
> even attempts to handle them as one might expect mathematically

XBRL precision means significant figures, and "-1 significant figures" has no meaning mathematically.  I appreciate the implementation does something, I just can't reason about these testcases.  Hopefully we don't let these undefined cases contort the code too badly.